sky/provision/fluidstack/instance.py

```diff
@@ -89,18 +89,47 @@ def run_instances(region: str, cluster_name_on_cloud: str,
 
     while True:
         instances = _filter_instances(cluster_name_on_cloud, pending_status)
+        if len(instances) > config.count:
+            raise RuntimeError(
+                f'Cluster {cluster_name_on_cloud} already has '
+                f'{len(instances)} nodes, but {config.count} are '
+                'required. Please try terminate the cluster and retry.')
         if not instances:
             break
         logger.info(f'Waiting for {len(instances)} instances to be ready.')
         time.sleep(POLL_INTERVAL)
     exist_instances = _filter_instances(cluster_name_on_cloud, ['running'])
     head_instance_id = _get_head_instance_id(exist_instances)
 
+    def rename(instance_id, new_name):
+        try:
+            utils.FluidstackClient().rename(instance_id, new_name)
+        except Exception as e:
+            logger.warning(f'run_instances error: {e}')
+            raise
+
+    for inst_id, inst in exist_instances.items():
+        if head_instance_id is None:
```

Comment:
leave a comment: when would it be the case that `exist_instances` does not contain any instance names with `-head` suffix?

Reply:
Added. Thanks!